src/helpers.js

```diff
@@ -0,0 +1,121 @@
+/**
+ *Created by Mikael Lindahl on 2018-01-05
+ */
+
+"use strict";
+
+const TYPES = {
+    number: 'INTEGER',
+    string: 'TEXT',
+    boolean: 'BOOLEAN',
+    json: 'JSON'
+}
+
+/**
+ * Crete model definitions to be used when defining tables
+ * with the dapteres
+ *
+ * @param models
+ * @param datastores
+ * @returns {{}}
+ */
+function createDefinition(models, datastores) {
+
+    let definition = {};
+
+    for (let key1 in models) {
+
+        let model = models[key1];
+        let adapter = datastores[model.datastore].adapter;
+
+        if (!definition[adapter]){
+
+            definition[adapter]={}
+
+        }
+
+        definition[adapter][key1]= {
+                config: JSON.parse(JSON.stringify(model.attributes)),
+                datastore: model.datastore
+            }
+
+
+        for (let key2 in model.attributes) {
+
+            definition[adapter][key1].config[key2].columnName = key2;
+
+            if (key2 == model.primaryKey) {
+
+                definition[adapter][key1].config[key2].primaryKey = true
+
+            }
+
+            if (model.attributes[key2].autoMigrations != {}) {
+                for (let key3 in model.attributes[key2].autoMigrations) {
+
+                    let migration = model.attributes[key2].autoMigrations[key3]
```

Comment:
With previous logic the expression would be shorter:
`let migration = autoMigrations[key3]`
Long expressions I would recommend to replace by let / const variables.